src/Controller/HomeController.php

```diff
@@ -14,22 +14,35 @@
  */
 class HomeController extends AbstractController
 {
+    /** @var MissionClient */
+    protected $missionClient;
+
+    public function __construct(MissionClient $missionClient)
+    {
+        $this->missionClient = $missionClient;
+    }
+
     /**
      * @Route("", name="_index")
      */
     public function indexAction(): Response
     {
-        $recentMissions = [];
-        for ($i = 1; $i < 7; ++$i) {
-            $recentMissions[] = [
-                'title' => "Mission {$i}",
-                'description' => 'Lorem ipsum, ArmaForces karamba.',
-                'date' => new DateTime("-{$i} day"),
-            ];
+        $nearestMission = $this->missionClient->getNearestMission();
+        $lastMission = null;
+
+        if (null === $nearestMission) {
```

Comment:
Some of this looping logic should be moved to MissionClient, some kind of utility class that would help with results processing or to Twig directly. Right now we have multiple calls to MissionClient service here and each of them can throw one or more exceptions and therefore exception handling directly in action will be difficult (and ugly).

We don't need any advanced error handling right now but it would be best to simply call two/three methods from here, wrapped in try/catch block with base Exception class. I think we can presume that if there will be any error in these calls we can just set results to null/empty arrays, log exception and display nothing. Simple and good enough for now.